File: android/apolloui/src/main/java/io/muun/apollo/presentation/ui/new_operation/NewOperationView.java
package io.muun.apollo.presentation.ui.new_operation;

import io.muun.apollo.domain.model.CurrencyDisplayMode;
import io.muun.apollo.domain.model.PaymentAnalysis;
import io.muun.apollo.domain.model.PaymentRequest;
import io.muun.apollo.presentation.ui.base.BaseView;

import androidx.annotation.NonNull;

public interface NewOperationView extends BaseView {

    void setCurrencyDisplayMode(CurrencyDisplayMode mode);

    void setForm(NewOperationForm form);

    void setPaymentAnalysis(PaymentAnalysis analysis);

    void setLoading(boolean isLoading);

    void setConnectedToNetwork(boolean isConnected);

    void editFee(@NonNull PaymentRequest paymentRequest);

    void editFeeManually(@NonNull PaymentRequest paymentRequest);

    void confirmFee(double selectedFeeRate);

    void showErrorScreen(NewOperationErrorType type);

}

File: android/apolloui/src/main/java/io/muun/apollo/presentation/ui/new_operation/NewOperationView.kt
package io.muun.apollo.presentation.ui.new_operation

import io.muun.apollo.domain.model.BitcoinUnit
import io.muun.apollo.domain.model.Contact
import io.muun.apollo.domain.model.SubmarineSwap
import io.muun.apollo.presentation.ui.base.BaseView
import newop.EnterAmountState
import newop.EnterDescriptionState

interface NewOperationView : BaseView {

    fun setInitialBitcoinUnit(bitcoinUnit: BitcoinUnit)

    fun setLoading(isLoading: Boolean)

    fun showErrorScreen(type: NewOperationErrorType)

    fun goToResolvingState()

    fun goToEnterAmountState(state: EnterAmountState, receiver: Receiver)

    fun setAmountInputError()

    fun goToEnterDescriptionState(state: EnterDescriptionState, receiver: Receiver)

    fun goToConfirmState(state: ConfirmStateViewModel, receiver: Receiver)

    fun goToEditFeeState()

    fun goToEditFeeManually()

    fun goToConfirmedFee()

    fun showAbortDialog()

    fun finishAndGoHome()

    interface Receiver {
        val swap: SubmarineSwap?
        val contact: Contact?
    }
}